adapters/Snap/SnapAdapter/GADMediationAdapterSnap.m

```diff
@@ -14,71 +14,112 @@
+      version.patchVersion = versionComponents[2].integerValue;
+    }
+    return version;
+}
+
++ (GADVersionNumber)adSDKVersion
+{
+    NSArray<NSString *> *versionComponents = [SAKMobileAd.shared.sdkVersion componentsSeparatedByString:@"."];
+    GADVersionNumber version = {0};
+    if (versionComponents.count == 3) {
+      version.majorVersion = versionComponents[0].integerValue;
+      version.minorVersion = versionComponents[1].integerValue;
+      version.patchVersion = versionComponents[2].integerValue;
+    }
+    return version;
 }
 
-+ (GADVersionNumber)adSDKVersion {
-  // TODO: Populate `versionString` with the ad-network's SDK version in NSString format.
-  NSString *versionString = @"";
-  GADVersionNumber version = {0};
-  NSArray<NSString *> *components = [versionString componentsSeparatedByString:@"."];
-  if (components.count == 3) {
-    version.majorVersion = components[0].integerValue;
-    version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue;
-  } else {
-    NSLog(@"Unexpected ad SDK version string: %@. Returning 0 for adSDKVersion.", versionString);
-  }
-  return version;
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass
+{
+    return nil;
 }
 
-+ (GADVersionNumber)adapterVersion {
-  GADVersionNumber version = {0};
-  NSArray<NSString *> *components = [GADMAdapterSnapVersion componentsSeparatedByString:@"."];
-  if (components.count == 4) {
-    version.majorVersion = components[0].integerValue;
-    version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue * 100 + components[3].integerValue;
-  }
-  return version;
++ (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
+             completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler
+{
+    NSMutableArray<NSString *> *snapAppIds = [NSMutableArray new];
+    for (GADMediationCredentials *credentials in configuration.credentials) {
+        NSString *snapAppId = credentials.settings[GADMAdapterSnapAppID];
+        if (snapAppId.length) {
+            [snapAppIds addObject:snapAppId];
+        }
+    }
+    
+    if (!snapAppIds.count) {
+        return completionHandler([NSError errorWithDomain:GADErrorDomain
+                                                     code:GADErrorInvalidRequest
+                                                 userInfo:@{ NSLocalizedDescriptionKey : @"Snap Audience Network requires a single snapAppId. No snapAppId were provided." }]);
+    }
+    if (snapAppIds.count > 1) {
+        return completionHandler([NSError errorWithDomain:GADErrorDomain
+                                                     code:GADErrorInvalidRequest
+                                                 userInfo:@{ NSLocalizedDescriptionKey : @"Snap Audience Network supports a single snapAppId but multiple IDs were provided." }]);
+    }
```

Comment:
My apology, it should just log a warning message instead of returning an error.

See below for reference:
https://github.com/googleads/googleads-mobile-ios-mediation/pull/333/files#diff-76d630cbf4e89e5fe3b1beee31073efbd49b49d73b4b71f633964e60016d3cceR52-R58